# Gap
### from "Men's Responsibly Made" get the links to each product

In [2]:
from selenium import webdriver
from time import sleep
import random
import pandas as pd

sleepTimes = [2.1, 2.8, 3.2]

url = "https://www.gap.com/browse/category.do?cid=1149858&nav=hamnav%3AMen%3A%3AResponsibly-Made%20Shop#pageId=0&department=75"
driver = webdriver.Firefox()

driver.get(url)
y = 1000
while y <= driver.execute_script("return document.body.scrollHeight"):
    driver.execute_script("window.scrollTo(0, "+str(y)+")")
    y += 1000
    sleep(1)

print("at bottom")

links = []
for card in driver.find_elements_by_class_name('product-card-grid__item-1-2'):
    try:
        link = card.find_element_by_tag_name('a').get_attribute("href")
        links.append(link)
    except:
        print(len(links))
        print(card.get_attribute('innerHTML'))

len(links)

at bottom


218

218 items

### visit each product link and get the material info

In [39]:
info_text = []
for link in links:
    driver.get(link)
    try:
        driver.find_element_by_css_selector('.promoDrawer__wrapper.is-expanded.promoDrawer__wrapper--opened').click()
    except:
        pass
    sleep(2)
    try:
        driver.find_element_by_class_name('sitewide-1qosac6').click()
        sleep(2)
    except:
        pass
    driver.find_element_by_xpath('//button[@class="pdp-drawer-trigger"][@aria-label="fabric & care - Opens Drawer"]').click()
    for span in driver.find_elements_by_xpath('//ul[@class="product-information-item__list"]/li/span'):
        if "%" in span.text:
            info_text.append(span.text)
        else:
            continue
    sleep(random.choice(sleepTimes))

len(info_text)

241

### save

In [40]:
materials_gap = pd.Series(info_text)
materials_gap.to_csv("materials_gap_men")

## Repeat Women

### from "Women's Responibly Made" get the links to get each product

In [42]:
url = "https://www.gap.com/browse/category.do?cid=1147561&nav=hamnav%3AWomen%3AJust%20Arrived%3AResponsibly-Made%20Shop%20"

driver.get(url)
y = 1000
while y <= driver.execute_script("return document.body.scrollHeight"):
    driver.execute_script("window.scrollTo(0, "+str(y)+")")
    y += 1000
    sleep(1)

print("at bottom")

links_women = []
for card in driver.find_elements_by_class_name('product-card-grid__item-1-2'):
    try:
        link = card.find_element_by_tag_name('a').get_attribute("href")
        links_women.append(link)
    except:
        print(len(links_women))
        print(card.get_attribute('innerHTML'))

len(links_women)

# second page
url = "https://www.gap.com/browse/category.do?cid=1147561&nav=meganav%3AWomen%3AJust%20Arrived%3AResponsibly-Made%20Shop%20#pageId=1"

driver.get(url)
y = 1000
while y <= driver.execute_script("return document.body.scrollHeight"):
    driver.execute_script("window.scrollTo(0, "+str(y)+")")
    y += 1000
    sleep(1)

for card in driver.find_elements_by_class_name('product-card-grid__item-1-2'):
    try:
        link = card.find_element_by_tag_name('a').get_attribute("href")
        links_women.append(link)
    except:
        print(len(links_women))
        print(card.get_attribute('innerHTML'))

len(links_women)

at bottom


453

453 items

### add Women's second page

### visit each product link and get the material info

In [44]:
info_text_women = []
for link in links_women:
    driver.get(link)
    try:
        driver.find_element_by_css_selector('.promoDrawer__wrapper.is-expanded.promoDrawer__wrapper--opened').click()
    except:
        pass
    sleep(2)
    try:
        driver.find_element_by_class_name('sitewide-1qosac6').click()
        sleep(2)
    except:
        pass
    driver.find_element_by_xpath('//button[@class="pdp-drawer-trigger"][@aria-label="fabric & care - Opens Drawer"]').click()
    for span in driver.find_elements_by_xpath('//ul[@class="product-information-item__list"]/li/span'):
        if "%" in span.text:
            info_text_women.append(span.text)
        else:
            continue
    sleep(random.choice(sleepTimes))

len(info_text_women)

505

### save

In [45]:
materials_gap_women = pd.Series(info_text_women)
materials_gap_women.to_csv("materials_gap_women")